File: src/main/java/com/paragon/api/event/player/RenderItemEvent.java
package com.paragon.api.event.player;

import me.wolfsurge.cerauno.event.Event;
import net.minecraft.util.EnumHandSide;

/**
 * @author Wolfsurge
 */
public class RenderItemEvent extends Event {

    private EnumHandSide side;

    public RenderItemEvent(EnumHandSide enumHandSide) {
        this.side = enumHandSide;
    }

    public EnumHandSide getSide() {
        return side;
    }

    public static class Pre extends RenderItemEvent {
        public Pre(EnumHandSide enumHandSide) {
            super(enumHandSide);
        }
    }

    public static class Post extends RenderItemEvent {
        public Post(EnumHandSide enumHandSide) {
            super(enumHandSide);
        }
    }

}

File: src/main/kotlin/com/paragon/api/event/player/RenderItemEvent.kt
package com.paragon.api.event.player

import com.paragon.bus.event.Event
import net.minecraft.util.EnumHandSide

/**
 * @author Surge
 */
open class RenderItemEvent(val side: EnumHandSide) : Event() {

    class Pre(enumHandSide: EnumHandSide) : RenderItemEvent(enumHandSide)
    class Post(enumHandSide: EnumHandSide) : RenderItemEvent(enumHandSide)

}